File: cpp_srccli/src/add_two_ints_server.cpp
#include <functional>
#include <memory>

#include "rclcpp/rclcpp.hpp"
#include "tutorial_interfaces/srv/add_three_ints.hpp"


void add(const std::shared_ptr<tutorial_interfaces::srv::AddThreeInts::Request> request, std::shared_ptr<tutorial_interfaces::srv::AddThreeInts::Response> response)
{
    response->sum = request->a + request->b + request->c;
    RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Incoming request\na: %ld" "b: %ld" "c: %ld", request->a, request->b, request->c);
    RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Sending back response: [%ld]" , (long int)response->sum);
}


int main(int args, char **argv)
{
    rclcpp::init(args,argv);
    std::shared_ptr<rclcpp::Node> node = rclcpp::Node::make_shared("add_two_ints_server");
    rclcpp::Service<tutorial_interfaces::srv::AddThreeInts>::SharedPtr service = node->create_service<tutorial_interfaces::srv::AddThreeInts>("add_two_ints", &add);
    RCLCPP_INFO(rclcpp::get_logger("rclcpp"),"Ready to add two ints.");
    rclcpp::spin(node);
    rclcpp::shutdown();
}

File: py_srvcli/py_srvcli/add_two_ints_server.py
import rclpy
from rclpy.node import Node

from example_interfaces.srv import AddTwoInts


class Server(Node):
    def __init__(self):
        super().__init__("add_two_ints_server")
        self.server_ = self.create_service(AddTwoInts, "add_two_ints", self.callback)

    def callback(self, request, response):
        response.sum = request.a + request.b
        self.get_logger().info("Incoming Request\na: %d b: %d" % (request.a,request.b))
        return response

def main(args=None):
    rclpy.init(args=args)
    server = Server()
    rclpy.spin(server)
    server.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()